apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts

```diff
@@ -72,10 +74,15 @@ export function InstructorQuestionSettings({
   origHash: string;
   canEdit: boolean;
   courseTopics: Topic[];
+  courseTags: Tag[];
 }) {
   // Only show assessments on which this question is used when viewing the question
   // in the context of a course instance.
   const shouldShowAssessmentsList = !!resLocals.course_instance;
+  let selectedTags: (string | null)[] = [];
+  if (resLocals.tags) {
+    selectedTags = resLocals.tags.map((tag) => tag.name);
+  }
```

Comment:
Given how this is used, it'd be moderately more efficient to use a `Set` instead of an array for `selectedTags`.

```suggestion
  const selectedTags = new Set(resLocals.tags?.map((tag) => tag.name) ?? []);
```

You can then use `selectedTags.has(...)` instead of `selectedTags.includes(...)` later on.

Reply:
Updated to use a `Set` in [d426d26](https://github.com/PrairieLearn/PrairieLearn/pull/10973/commits/d426d26b01edba612257c99691275d279950c468).